service/src/main/java/com/ge/predix/acs/zone/management/dao/ZoneEntity.java

```diff
@@ -155,19 +152,27 @@ public String toString() {
                 + ", subdomain=" + this.subdomain + "]";
     }
 
-    public void setResourceAttributeConnector(final AttributeConnectorEntity connector) {
-        this.resourceAttributeConnector = connector;
+    public void setResourceAttributeConnector(final AttributeConnector connector) throws JsonProcessingException {
+        this.resourceAttributeConnector = OBJECT_MAPPER.writeValueAsString(connector);
     }
 
-    public AttributeConnectorEntity getResourceAttributeConnector() {
-        return this.resourceAttributeConnector;
+    public AttributeConnector getResourceAttributeConnector()
+            throws JsonParseException, JsonMappingException, IOException {
+        if (null == this.resourceAttributeConnector) {
+            return null;
+        }
+        return OBJECT_MAPPER.readValue(this.resourceAttributeConnector, AttributeConnector.class);
```

Comment:
this method should deserialize the json only when the json has changed 
* keep a separate field of type AttributeConnector
* initialize from josn when getter is called, only if null
* setter should also set this field, and serialize to json only if new value  does not equal current value in object